tldr/semantic.py

```
@@ -733,6 +733,114 @@ def _process_file_for_extraction(
+                def find_parent_class(node, class_name=None):
+                    """Find parent class by walking up from method node."""
+                    if node is None or node.id == root_node.id:
+                        return class_name
+                    parent = node.parent
+                    if parent and parent.type == "class_declaration":
+                        for child in parent.children:
+                            if child.type == "name":
+                                return content[child.start_byte:child.end_byte]
+                    return find_parent_class(parent, class_name)
+
+                def extract_code_preview(node, start_line):
```

Ruff (0.14.13)

Missing return type annotation for private function `extract_code_preview`

(ANN202)